xs/src/libslic3r/GCode.cpp

```diff
@@ -508,9 +508,12 @@ GCode::_extrude(ExtrusionPath path, std::string description, double speed)
     }
     
     // compensate retraction
-    gcode += this->unretract();
+	if (!this->writer.extruder()->is_constant_rate())
+		gcode += this->unretract();
     
     // adjust acceleration
+	// Don't adjust if using a constant extruder
+	if (!this->writer.extruder()->is_constant_rate())
```

Comment:
Is this actually needed? User can just leave acceleration settings to 0 and Slic3r will not emit any acceleration command.